tfx/dsl/component/experimental/annotations.py

```diff
@@ -17,10 +17,18 @@
 """
 
 import inspect
+import sys
+
 from typing import Any, Type, Union, Dict, List
 
 from tfx.dsl.component.experimental import json_compat
 from tfx.types import artifact
+
+if sys.version_info >= (3, 8):
```

Comment:
I guess this should be done from typing_extensions, and I think we're already supporting only python>=3.8 so don't need this.